shared_model/interfaces/iroha_internal/transaction_batch.hpp

```diff
@@ -0,0 +1,94 @@
+#define IROHA_TRANSACTION_BATCH_HPP
+
+#include "common/result.hpp"
+#include "interfaces/common_objects/transaction_sequence_common.hpp"
+#include "validators/transactions_collection/transactions_collection_validator.hpp"
+
+namespace shared_model {
+  namespace interface {
+
+    class TransactionBatch {
+     public:
+      TransactionBatch() = delete;
+
+      /**
+       * Create transaction batch out of collection of transactions
+       * @tparam TransactionValidator validates every single transaction
+       * @tparam OrderValidator validates order of transactions
+       * @param transactions collection of transactions, should be from the same
+       * batch
+       * @param validator transactions collection validator with provided
+       * transaction validator and order validator
+       * @return valid batch of transactions
+       */
+      template <typename TransactionValidator, typename OrderValidator>
+      static iroha::expected::Result<TransactionBatch, std::string>
+      createTransactionBatch(const types::SharedTxsCollectionType &transactions,
+                             const validation::TransactionsCollectionValidator<
+                                 TransactionValidator,
+                                 OrderValidator> &validator);
+
+      /**
+       * Creates transaction batch from single transaction
+       * @tparam TransactionValidator validates every single transaction
+       * @param transaction is transaction being validated and used to create
+       * batch
+       * @param transaction_validator transaction validation logic
+       * @return batch with single transaction
+       * @note transactions in such batches may not have batch meta information
+       */
+      template <typename TransactionValidator>
+      static iroha::expected::Result<TransactionBatch, std::string>
+      createTransactionBatch(std::shared_ptr<Transaction> transaction,
+                             const TransactionValidator &transaction_validator =
+                                 TransactionValidator());
+
+      /**
+       * Get transactions list
+       * @return list of transactions from the batch
+       */
+      const types::SharedTxsCollectionType &transactions() const;
+
+      /**
+       * Get the concatenation of reduced hashes as a single hash
+       * @param reduced_hashes collection of reduced hashes
+       * @return concatenated reduced hashes
+       */
+      const types::HashType &reducedHash() const;
+
+      /**
+       * Checks if every transaction has quorum signatures
+       * @return true if every transaction has quorum signatures, false
+       * otherwise
+       */
+      bool hasAllSignatures() const;
+
+      /**
+       * Get the concatenation of reduced hashes as a single hash
+       * That kind of hash does not respect batch type
+       * @param reduced_hashes
+       * @return concatenated reduced hashes
+       */
+      static types::HashType calculateReducedBatchHash(
```

Comment:
Why is this method public?

Reply:
Because it is used in transaction sequence as well